ecc/bls12-381/g1.go

```diff
@@ -421,8 +428,15 @@ func (p *G1Jac) mulWindowed(a *G1Jac, s *big.Int) *G1Jac {
 	var res G1Jac
 	var ops [3]G1Jac
 
-	res.Set(&g1Infinity)
 	ops[0].Set(a)
+	e := s
+	if s.Sign() == -1 {
+		e = bigIntPool.Get().(*big.Int)
+		defer bigIntPool.Put(e)
+		e.Neg(s)
+		ops[0].Neg(&ops[0])
+	}
+	res.Set(&g1Infinity)
```

Comment:
Just randomly browsed this particular thing. Looks like a lot of code with no side-effect. `e.Neg(s)` set `e` as `-s`, but otherwise is side-effect-free. And that's the last we see of `e`. So this whole chunk can be replaced by 
```golang
	if s.Sign() == -1 {
		ops[0].Neg(&ops[0])
	}
	res.Set(&g1Infinity)
```
Curius to know if I'm wrong?